schemas/2015-08-31-preview/Microsoft.ManagedIdentity.json

```diff
@@ -33,22 +33,41 @@
               "type": "object",
               "additionalProperties": {
                 "type": "string"
-              }
+              },
+              "properties": {}
             },
             {
-              "$ref": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#/definitions/expression"
+              "$ref": "https://schema.management.azure.com/schemas/common/definitions.json#/definitions/expression"
             }
           ],
           "description": "Resource tags"
+        },
+        "properties": {
+          "oneOf": [
+            {
+              "$ref": "#/definitions/IdentityProperties"
+            },
+            {
+              "$ref": "https://schema.management.azure.com/schemas/common/definitions.json#/definitions/expression"
+            }
+          ],
+          "description": "The properties associated with the identity."
         }
       },
       "required": [
         "name",
         "type",
         "apiVersion",
-        "location"
```

Comment:
Why is location being removed from the required properties?

Reply:
It's not been marked as required in your swagger definition:
https://github.com/Azure/azure-rest-api-specs/blob/master/specification/msi/resource-manager/Microsoft.ManagedIdentity/stable/2018-11-30/ManagedIdentity.json#L333-L336

Reply:
Seems like this "location" is still being removed. However, we marked it as required here: 
https://github.com/Azure/azure-rest-api-specs/blob/21460cb6be715578c300a130b9af7acbf277fbc8/specification/msi/resource-manager/Microsoft.ManagedIdentity/stable/2018-11-30/ManagedIdentity.json#L402

How else can we mark it as required?

Reply:
If I move the "required" field from the [Identity definition](https://github.com/Azure/azure-rest-api-specs/blob/21460cb6be715578c300a130b9af7acbf277fbc8/specification/msi/resource-manager/Microsoft.ManagedIdentity/stable/2018-11-30/ManagedIdentity.json#L402-L404) to the [Resource definition](https://github.com/Azure/azure-rest-api-specs/blob/21460cb6be715578c300a130b9af7acbf277fbc8/specification/msi/resource-manager/Microsoft.ManagedIdentity/stable/2018-11-30/ManagedIdentity.json#L371), the schema is correctly generated with location as required.

Reply:
Ok, we have a PR that should address that, using Azure common types: https://github.com/Azure/azure-rest-api-specs/pull/8352/

Reply:
@anthony-c-martin can you try to regenerate? We've updated our swagger to use Azure common types definitions.